AudioPlayer/AudioPlayer/player/AudioPlayerDelegate.swift

```diff
@@ -13,7 +13,7 @@ public typealias Metadata = [AVMetadataItem]
 
 /// This protocol contains helpful methods to alert you of specific events. If you want to be notified about those
 /// events, you will have to set a delegate to your `audioPlayer` instance.
-public protocol AudioPlayerDelegate: class {
+@objc public protocol AudioPlayerDelegate {
```

Comment:
What happens to the extension of the protocol to make funcs optional here? Should we also mark each func as `optional`?